File: Data Structures & Algorithms/balanced-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def dfs(self, root: Optional[TreeNode]) -> int:
            if not root:
                return 0
            
            return 1 + max(self.dfs(root.right), self.dfs(root.left))
    
    def isBalanced(self, root: Optional[TreeNode]) -> bool:

        if not root:
            return True

        left = self.dfs(root.left)
        right = self.dfs(root.right)

        if abs(left - right) > 1:
            return False
        
        return self.isBalanced(root.left) and self.isBalanced(root.right)

            

File: Data Structures & Algorithms/is-palindrome/submission-0.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        
    string str = "";    

    if (s.size() == 0 || s.size() == 1){
        return true;
    }

    for(const auto& elem : s){
        if (isalnum(elem)){
            str += tolower(elem);
        }
    }

    int left = 0;
    int right = str.size() - 1;

    while (left < right) {
        if (str[left] != str[right]){
            return false;
        }    
        left++;
        right--;
    }
    return true;
}
};
